src/persist-client/src/write.rs

```diff
@@ -454,9 +454,16 @@ where
         let desc = Description::new(lower, upper, since);
 
         let (mut parts, mut num_updates) = (Vec::new(), 0);
+        let mut runs = vec![];
         for batch in batches.iter() {
             let () = validate_truncate_batch(&batch.batch.desc, &desc)?;
-            parts.extend_from_slice(&batch.batch.parts);
+            for run in batch.batch.runs() {
+                // Mark the boundary if this is not the first run in the batch.
+                if parts.len() != 0 {
```

Comment:
I remembered why I did it this way, which is that I found it more clear that we're skipping the implicit run start at zero. So I've made that explicit via naming instead of calling len twice. Hope that's fine!